src/util/logging.ts

```diff
@@ -0,0 +1,77 @@
+export enum LogLevel {
+  TRACE = 0,
+  DEBUG = 1,
+  INFO = 2,
+  WARN = 3,
+  ERROR = 4,
+}
+
+/**
+ * Converts the FAUNA_DEBUG environment variable (string) into a LogLevel.
+ *   The intended use is to set FAUNA_DEBUG=0|1.
+ *
+ *   This function will convert null, undefined, empty, or or any string
+ *   starting with "0" to false.
+ *
+ * @param fauna_debug - The String value of FAUNA_DEBUG.
+ */
+export function faunaDebugEnabled(fauna_debug: string | undefined): boolean {
+  if (fauna_debug === undefined || fauna_debug === null) {
+    return false;
+  } else {
+    if (fauna_debug.length == 0) {
+      return false;
+    } else {
+      if (fauna_debug.startsWith("0")) {
+        return false;
+      } else {
+        return true;
+      }
+    }
+  }
+}
+
+export interface LogHandler {
+  trace(msg?: string, args?: string[]): void;
+  debug(msg?: string, args?: string[]): void;
+  info(msg?: string, args?: string[]): void;
+  warn(msg?: string, args?: string[]): void;
+  error(msg?: string, args?: string[]): void;
+}
+
+export class ConsoleLogHandler implements LogHandler {
+  constructor(private readonly level: LogLevel) {
```

Comment:
prefer native private class properties over typescript magic ones. Declaring the field above the constructor will also ensure at a type level that it is assigned (which is currently not guaranteed).
```suggestion
  readonly #level: LogLevel;
  constructor(level: LogLevel) {
```